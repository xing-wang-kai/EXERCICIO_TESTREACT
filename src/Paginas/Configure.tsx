import Formulario from "../components/formulario/Formulario";
import ParticipantesList from "../components/ParticipanteList/ParticipantesList";
import Rodape from "../components/Rodape/Rodape";
import Card from '../components/Card/card';


export default function Configure () {
    return(
        <Card>
            <section>
                <h2>Let's to play!</h2>
                <Formulario/>
                <ParticipantesList/>
                <Rodape/>
            </section>
        </Card>
    )
}