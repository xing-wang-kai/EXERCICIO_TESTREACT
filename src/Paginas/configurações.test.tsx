import React from 'react';
import { render } from '@testing-library/react';
import { RecoilRoot } from 'recoil';
import Configure from './Configure';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => {
    return {
        useNavigate: ()=> mockNavigate
    }
});

describe('GROUP:--> The Configure Page:', () => {
  it("TEST:--> needs to be rendered correctly:", ()=>{
    const { container } = render(<RecoilRoot><Configure/></RecoilRoot>)

    expect( container ).toMatchSnapshot();

  })

  
})
 