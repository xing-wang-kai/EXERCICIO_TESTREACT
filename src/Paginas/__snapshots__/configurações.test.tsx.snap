// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GROUP:--> The Configure Page: TEST:--> needs to be rendered correctly: 1`] = `
<div>
  <div
    class="sc-ftTHYK jMnoeW"
  >
    <section>
      <h2>
        Let's to play!
      </h2>
      <form>
        <div
          class="sc-bcXHqe jXSIqJ"
        >
          <label
            for="nome"
          />
          <input
            class="sc-gswNZR bFSTTR nome"
            id="nome"
            placeholder="Insira o nome dos participantes"
            required=""
            type="text"
            value=""
          />
          <button
            class="sc-dkrFOg jiHJa-d"
            disabled=""
          >
            Adicionar! 
          </button>
        </div>
        
      </form>
      <div>
        <h3>
          NOMES NA LISTA DE PARTICIPANTES: 
        </h3>
        <ul />
      </div>
      <div>
        <footer
          class="sc-eDvSVe gldbwB"
        >
          <button
            class="sc-jSUZER EQdME"
            disabled=""
          >
             Come√ßar a Brincadeira
          </button>
          <img
            alt="Sacolas de compras"
            src="/imagens/sacolas.png"
          />
        </footer>
        <br />
        <hr />
        <h4
          class="sc-iBYQkv cQYBaA"
        >
          Feito com s2 por Kai wang
        </h4>
      </div>
    </section>
  </div>
</div>
`;
